src/common/hooks/useOnClickOutside/useOnClickOutside.ts

```diff
@@ -0,0 +1,18 @@
+import { RefObject, useEffect } from 'react';
+
+export const useOnClickOutside = (ref: RefObject<HTMLElement>, handler: (event: MouseEvent) => void) => {
```

Comment:
```suggestion
export const useOnClickOutside = <T extends HTMLElement = HTMLElement>( ref: RefObject<T>, handler: (event: MouseEvent | TouchEvent) => void ) => {
```


By using <T extends HTMLElement = HTMLElement>, This allows the hook to be more flexible, as it can now be used with any specific type of HTMLElement, not just the generic HTMLElement. This is useful when you know the specific type of element you'll be working with and want to ensure type safety throughout your component.